### ISSUE TITLE
node-lambda incompatible with AWS X-Ray

### ISSUE THREAD

**Issue by vladaman** [motdotla/node-lambda#288]
Does anyone know, how to fix issue with node-lambda run and AWS X-Ray? When I include following:

```
var AWSXRay = require('aws-xray-sdk-core');
var AWS = AWSXRay.captureAWS(require('aws-sdk'));
```
I get this error:

```
/lambda/node_modules/aws-xray-sdk-core/lib/context_utils.js:34
      throw new Error(message);
      ^
Error: Failed to get the current sub/segment from the context.
```

**Comment by RickDT** [comment]
You can prepend the `run` command with an ENV var that disables it:
`"run": "AWS_XRAY_CONTEXT_MISSING=LOG_ERROR node-lambda run"`

**Comment by DeviaVir** [comment]
Closing due to no response registered and a possible solution offered. Feel free to comment in case the issue persisted.

**Comment by jeanguerin** [comment]
Hello, 

Thank you for this solution which work quite well. 
I have add 
`"AWS_XRAY_CONTEXT_MISSING=LOG_ERROR" >> .env`
and 
```
var AWSXRay = require('aws-xray-sdk-core');
var AWS = AWSXRay.captureAWS(require('aws-sdk'));
```
works just fine.

Now I would like to go further using subsegment, for example for capturing annotations.
```
AWSXRay.captureFunc('annotations', function(subsegment){
    subsegment.addAnnotation('Name', name);
    subsegment.addAnnotation('UserID', event.userid);
  });
```
I works locally with Mocha. I works when deployed on AWS. But 
node-lambda run returns
```
      subsegment.addAnnotation('Name', "name");
                  ^
TypeError: Cannot read property 'addAnnotation' of undefined
```
I think, it doesn't find any segment. Could you help me please?

Best,

JG

**Comment by abetomo** [comment]
@jeanguerin What version of `node-lambda` do you use?

**Comment by abetomo** [comment]
@jeanguerin Could you paste an error displayed in 0.11.6?
I want to know what line of `node-lambda` caused an error.

**Comment by jeanguerin** [comment]
Tell me if you need another log.
```
TypeError: Cannot read property 'addAnnotation' of undefined
    at /backend/administration/testFunction.js:48:19
    at Object.captureFunc (/backend/lib/node_modules/aws-xray-sdk-core/lib/capture.js:27:12)
    at Object.module.exports. testFunction (/backend/administration/testFunction.js:47:17)
    at module.exports.handler_ testFunction (/backend/administration/testFunction.js:31:10)
    at Lambda._runHandler (/usr/local/lib/node_modules/node-lambda/lib/main.js:119:5)
    at Lambda.run (/usr/local/lib/node_modules/node-lambda/lib/main.js:81:10)
    at Command.program.command.alias.description.option.option.option.option.option.option.option.option.action (/usr/local/lib/node_modules/node-lambda/bin/node-lambda:133:27)
    at Command.listener (/usr/local/lib/node_modules/node-lambda/node_modules/commander/index.js:315:8)
    at emitTwo (events.js:106:13)
    at Command.emit (events.js:191:7)
    at Command.parseArgs (/usr/local/lib/node_modules/node-lambda/node_modules/commander/index.js:651:12)
    at Command.parse (/usr/local/lib/node_modules/node-lambda/node_modules/commander/index.js:474:21)
    at Object.<anonymous> (/usr/local/lib/node_modules/node-lambda/bin/node-lambda:144:4)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3

```

**Comment by abetomo** [comment]
@jeanguerin Thank you.
I think that I understood the cause, so I will fix it.

**Comment by abetomo** [comment]
@jeanguerin Fixed #414

**Comment by jeanguerin** [comment]
Tested and approved. Thank you for your reactivity.
